tests/test_zooniverseCutouts.py

```diff
@@ -0,0 +1,129 @@
+                                                  self.skyCenter)
+        im = PIL.Image.open(cutout)
+        # NOTE: uncomment this to show the resulting image.
+        # im.show()
+        self.assertEqual(im.height, 233)
+        self.assertEqual(im.width, 630)
+
+        # A different cutout size: the resulting cutout image is the same
+        # size but shows more pixels.
+        config = ZooniverseCutoutsTask.ConfigClass()
+        config.size = 100
+        zooniverseCutouts = ZooniverseCutoutsTask(config=config)
+        cutout = zooniverseCutouts.generate_image(self.science, self.template, self.difference,
+                                                  self.skyCenter)
+        im = PIL.Image.open(cutout)
+        # NOTE: uncomment this to show the resulting image.
+        # im.show()
+        self.assertEqual(im.height, 233)
+        self.assertEqual(im.width, 630)
+
+    def test_generate_images(self):
+        """Test that images get written to a temporary directory."""
+        data = pd.DataFrame(data={"diaSourceId": [5, 10],
+                                  "ra": [45.001, 45.002],
+                                  "decl": [45.0, 45.001],
+                                  "ccd": [50, 60],
+                                  "visit": [1234, 5678]})
+        butler = unittest.mock.Mock(spec=lsst.daf.butler.Butler)
+        # we don't care what the output images look like here, just that
+        # butler.get() returns an Exposure for every call.
+        butler.get.return_value = self.science
+
+        with tempfile.TemporaryDirectory() as path:
+            config = ZooniverseCutoutsTask.ConfigClass()
+            config.outputPath = path
+            zooniverseCutouts = ZooniverseCutoutsTask(config)
+            zooniverseCutouts.generate_images(data, butler)
+            image = PIL.Image.open(os.path.join(path, "images/5.png"))
+            self.assertEqual(image.format, "PNG")
+            PIL.Image.open(os.path.join(path, "images/10.png"))
+            self.assertEqual(image.format, "PNG")
+
+    def check_make_manifest(self, url_root):
+        """Check that make_manifest returns an appropriate DataFrame.
+        """
+        data = pd.DataFrame(data={"diaSourceId": [5, 10, 20], "otherField": [3, 2, 1]})
+        config = ZooniverseCutoutsTask.ConfigClass()
+        config.urlRoot = url_root
+        zooniverseCutouts = ZooniverseCutoutsTask(config=config)
+        manifest = zooniverseCutouts.make_manifest(data)
+        self.assertEqual(manifest['metadata:diaSourceId'].to_list(),
+                         [5, 10, 20])
+        return manifest
+
+    def test_make_manifest(self):
+        # check with an ending slash
+        root = "http://example.org/zooniverse"
+        manifest = self.check_make_manifest(root)
+        self.assertEqual(manifest['location:1'].to_list(),
+                         [f"{root}/5.png",
+                          f"{root}/10.png",
+                          f"{root}/20.png"])
```

Comment:
I'm confused by the division of labor here. Why does `check_make_manifest` do some checks, and the caller others?

Reply:
I added a `url_list` arg to `check_make_manifest` and move the `assertEqual` into it.